### ISSUE TITLE
False positive `super-init-not-called` for subclasses of subclasses of subclasses of typing.Protocol

### ISSUE THREAD

**Issue by abulenok** [PyCQA/pylint#9225]
### Bug description

related to https://github.com/pylint-dev/pylint/issues/4790

Even though no `__init__` methods are implemented, `pylint` produces the `super-init-not-called` warning.

```python
from typing import Protocol

class TestProto(Protocol):
    """Note that there's no __init__ method here"""

class Test1(TestProto):
    """Note that there's no __init__ method here"""

class Test2(Test1):
    """Inherit explicitly from Test1"""

    def __init__(self):
        """Initialize this object"""
```

### Command used

```shell
pylint some.py
```

### Pylint output

```python
************* Module some
some.py:12:4: W0231: __init__ method from base class 'Test1' is not called (super-init-not-called)
```

### Expected behavior

No `super-init-not-called` warning produced

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]
```
